packages/grid/x-data-grid/src/colDef/gridSelectColDefUtils.tsx

```diff
@@ -0,0 +1,14 @@
+import { ValueOptions } from '../models/colDef/gridColDef';
+import { isObject } from '../utils/utils';
+
+export const isArrayOfObjects = (options: any): options is Array<Record<string, any>> => {
+  return typeof options[0] === 'object';
+};
```

Comment:
`typeof null === 'object'`, so the check isn't entirely correct for the typings declared here. Accessing `[0]` directly also means it's assumed that `options` must be non-null, which isn't obvious to the caller of this function due to the signature. In general for type assertion functions I prefer to type the argument as `unknown` rather than `any`.

Reply:
Totally agree.

Reply:
Went ahead and took the defensive approach.